File: problems/intermediate/organizing_container/__tests__/test_organizing_container.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include "../organizing_container.c"

void test() {
    int n = 2;
    int** containers = malloc(n * sizeof(int*));
    containers[0] = (int[]){1, 4};
    containers[1] = (int[]){2, 3};

    char* result = organizingContainers(containers, n);
    if (strcmp(result, "Impossible") != 0) {
        printf("❌ Test failed. Expected Impossible, got %s\n", result);
        exit(1);
    }

    printf("✅ C test passed!\n");
}

int main() {
    test();
    return 0;
}

File: problems/intermediate/organizing_container/__tests__/test_organizing_container.js
const { organizingContainers } = require("../organizing_containers");

const containers = [[1, 4], [2, 3]];
const result = organizingContainers(containers);

if (result !== "Impossible") {
    throw new Error(`❌ Test failed. Expected Impossible, got ${result}`);
}

console.log("✅ JavaScript test passed!");
